LLVM_full_jll does not contain clang++

To use LLVM 10 to build `julia`, I needed to add a symlink `clang++` manually like this https://github.com/JuliaLang/julia/issues/35341#issuecomment-609130650:

```sh
tooldir=$HOME/.julia/artifacts/8cd56e3e81e5770720c31304026f3b23f129153e/tools
cd $tooldir
ln -s clang-10 clang++
cd -
... PATH="$tooldir:$PATH" make USECLANG=1 ...
```

Is it intentional that `LLVM_full_jll` contains a symlink `clang` but not `clang++`?  If not, can this be added to the distribution?

> `clang++` vs `clang` makes no difference in my experience.

It does change the default language selection:

```
tim@taurus /tmp$ cat test.h                                                                                                                                                                                                                                                   
#ifdef __cplusplus
#error C++
#else
#error C
#endif

tim@taurus /tmp$ clang test.h                                                                                                                                                                                                                                                 
test.h:4:2: error: C
#error C
 ^
1 error generated.

tim@taurus /tmp$ clang++ test.h                                                                                                                                                                                                                                               
clang-9: warning: treating 'c-header' input as 'c++-header' when in C++ mode, this behavior is deprecated [-Wdeprecated]
test.h:2:2: error: C++
#error C++
 ^
1 error generated
```

> But we don't do anything different from the upstream project here.

When building LLVM/Clang with the Julia build system but not using BB we do get a symlink:

```
$ ls -la ./deps/llvm-6.0.1/build_Release/bin/clang++
lrwxrwxrwx 1 maleadt maleadt 9 Mar 29 16:19 ./deps/llvm-6.0.1/build_Release/bin/clang++ -> clang-6.0
```

And this seems to be done by upstream: https://github.com/llvm/llvm-project/blob/717a460867633008520713614018f3c37503a5ca/clang/tools/driver/CMakeLists.txt#L65-L71

So for consistency, and to make the LLVM_full tarball as usable as possible as a toolchain outside of Julia, it would be useful if it included these symlinks.

--- a/L/LLVM/LLVM_full@8.0.1/build_tarballs.jl
+++ b/L/LLVM/LLVM_full@8.0.1/build_tarballs.jl
@@ -2,6 +2,5 @@ version = v"8.0.1"
 
 include("../common.jl")
 
-
 build_tarballs(ARGS, configure_build(ARGS, version)...;
                      preferred_gcc_version=v"7", preferred_llvm_version=v"8")

--- a/L/LLVM/LLVM_full@9.0.1/build_tarballs.jl
+++ b/L/LLVM/LLVM_full@9.0.1/build_tarballs.jl
@@ -2,6 +2,5 @@ version = v"9.0.1"
 
 include("../common.jl")
 
-
 build_tarballs(ARGS, configure_build(ARGS, version)...;
                      preferred_gcc_version=v"7", preferred_llvm_version=v"8")

--- a/L/LLVM/common.jl
+++ b/L/LLVM/common.jl
@@ -139,11 +139,10 @@ CMAKE_FLAGS+=(-DCMAKE_CROSSCOMPILING=True)
 
 # Julia expects the produced LLVM tools to be installed into tools and not bin
 # We can't simply move bin to tools since on MingW64 it will also contain the shlib.
-CMAKE_FLAGS+=(-DLLVM_TOOLS_BINARY_DIR="\${LLVM_INSTALL_PREFIX}/tools")
-CMAKE_FLAGS+=(-DLLVM_TOOLS_INSTALL_DIR="\${LLVM_INSTALL_PREFIX}/tools")
+CMAKE_FLAGS+=(-DLLVM_TOOLS_INSTALL_DIR="tools")
 
 # Also build and install utils, since we want FileCheck, and lit
-CMAKE_FLAGS+=(-DLLVM_UTILS_INSTALL_DIR="\${LLVM_INSTALL_PREFIX}/tools")
+CMAKE_FLAGS+=(-DLLVM_UTILS_INSTALL_DIR="tools")
 CMAKE_FLAGS+=(-DLLVM_INCLUDE_UTILS=True -DLLVM_INSTALL_UTILS=True)
 
 # Include perf/oprofile/vtune markers

